### PR TITLE
15092 - remove fee summary components from App.vue add to each view instead

### PR DESCRIPTION
**Pull request by jonathan-longe**
*Issue #:* /bcgov/entity#15092

This is an incremental PR as the changes are potentially disruptive.  The PR removes the fee summaries from App.vue and moves them to a ViewWrapper component that's added to the following views:

- Alteration.vue
- Change.vue
- Conversion.vue
- Correction.vue
- Restoration.vue
- SpecialResolution.vue

By removing the fee summary components from App.vue it gives us more flexibility to handle the actions from FeeSummaryShared differently.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/ViewWrapper.vue [hunk 1]
```diff
@@ -0,0 +1,387 @@
+<template>
+  <div>
+    <v-container class="view-container my-8 py-0">
+      <v-row>
+        <v-col cols="9" class="left-side">
+          <slot></slot>
+        </v-col>
```
Comment (jonathan-longe) [inline src/components/ViewWrapper.vue:7]: The ViewWrapper takes a significant chunk of the template from App.vue.  Here we're using a `<slot>` to allow the views to add back their own content with the minimum of disruption.

File: src/views/Alteration.vue [hunk 2]
```diff
@@ -1,111 +1,113 @@
...
-          :sectionNumber="showTransactionalFolioNumber ? '3.' : '2.'"
-          :validate="getAppValidate"
-          :disableEdit="!isRoleStaff"
-        />
-
-        <!-- STAFF ONLY: Court Order/Plan of Arrangement and Staff Payment -->
-        <template v-if="isRoleStaff">
-          <CourtOrderPoa
+  <ViewWrapper>
+    <section class="pb-10" id="alteration-view">
```
Comment (jonathan-longe) [inline src/views/Alteration.vue:3]: The wrapper is added in each view like this.

File: src/views/Alteration.vue [hunk 3]
```diff
@@ -1,111 +1,113 @@
...
-          <StaffPayment
+          <DocumentsDelivery
             class="mt-10"
-            :sectionNumber="showTransactionalFolioNumber ? '5.' : '4.'"
-            @haveChanges="onStaffPaymentChanges()"
+            sectionNumber="1."
+            :validate="getAppValidate"
+            @valid="setDocumentOptionalEmailValidity($event)"
           />
-        </template>
```
Comment (jonathan-longe) [inline src/views/Alteration.vue:80]: This hasn't changed.  It's just Github being dumb with how it recognizes changes.

File: tests/unit/App.spec.ts [hunk 4]
```diff
@@ -651,20 +651,4 @@ describe('App component - other', () => {
...
-    store.state.stateModel.filingData = {
-      filingTypeCode: 'FMCHANGE',
-      entityType: 'SP',
-      priority: false,
-      waiveFees: false
-    }
-    await Vue.nextTick()
-
-    expect(wrapper.findComponent(FeeSummaryShared).exists()).toBe(true) // not displayed initially
-  })
```
Comment (jonathan-longe) [inline tests/unit/App.spec.ts:669]: Removed this test as FeeSummaryShared is no longer rendered with App.vue

**Review by severinbeauvais — COMMENTED** [review]
LGTM! Nice and concise!

**Review by ketaki-deodhar — APPROVED** [review]
Looks good. Clean!

**Review by JazzarKarim — APPROVED** [review]
Love it. No complaints here!

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**